src/pages/RouteDetails/components/Map/RoutingComponent.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useEffect } from 'react';
+import { useMap } from 'react-leaflet';
+import L from 'leaflet';
+import 'leaflet-routing-machine';
+
+import { store } from '@/store/store';
+import { setisVisible } from '@/store/slices/loaderSlice';
+
+interface RoutingComponentProps {
+  locations: string[];
+}
+
+const RoutingComponent: React.FC<RoutingComponentProps> = ({ locations }) => {
+  const map = useMap();
+
+  useEffect(() => {
+    let routingControl: L.Routing.Control;
+
+    const setupRoute = async (): Promise<void> => {
+      try {
+        store.dispatch(setisVisible(true));
+        const coordinates = await Promise.all([...locations].map(geocode));
+
+        routingControl = L.Routing.control({
+          waypoints: coordinates.map((coords) =>
+            L.latLng(coords.lat, coords.lon)
+          ),
+          routeWhileDragging: true,
+          createMarker: () => null,
+          lineOptions: {
+            styles: [
+              {
+                color: generateRandomColor(),
+                weight: 4,
+              },
+            ],
+          },
+          show: false,
+        }).addTo(map);
+      } catch (error) {
+        console.error('Failed to calculate route:', error);
```

Comment:
no console log, error should be shown to user

Reply:
fixed